sdk/eidos_sdk/builtins/logic_units/web_search.py

```diff
@@ -0,0 +1,115 @@
+import os
+from contextlib import asynccontextmanager
+from typing import List, Optional
+
+# from aiohttp import ClientSession
+from bs4 import BeautifulSoup
+from httpx import AsyncClient, Timeout
+from jinja2 import Environment, StrictUndefined
+from pydantic import BaseModel, Field
+
+from eidos_sdk.agent.client import Program
+from eidos_sdk.cpu.call_context import CallContext
+from eidos_sdk.cpu.logic_unit import LogicUnit, llm_function
+from eidos_sdk.cpu.memory_unit import MemoryUnit
+from eidos_sdk.system.reference_model import Specable
+from eidos_sdk.system.request_context import RequestContext
+from eidos_sdk.util.logger import logger
+
+
+class SearchResult(BaseModel):
+    url: str
+    title: str
+    description: str
+
+
+# Requires custom search engine + token setup in google project. See more at https://developers.google.com/custom-search/v1/reference/rest/v1/cse/list
+class WebSearchConfig(BaseModel):
```

Comment:
basic web search tool using google search.

I found that by extracting pure text from the html pages, the size is pretty small even for larger pages. We could introduce summarizer concepts, but I don't think they are needed for now.